server/issue_test.go

```diff
@@ -194,3 +205,172 @@ func TestRouteIssueTransition(t *testing.T) {
 	}
 
 }
+
+func TestRouteAttachCommentToIssue(t *testing.T) {
+	api := &plugintest.API{}
+
+	api.On("LogError",
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string")).Return(nil)
+
+	api.On("LogDebug",
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string"),
+		mock.AnythingOfTypeArgument("string")).Return(nil)
+
+	siteURL := "https://somelink.com"
+	api.On("GetConfig").Return(&model.Config{ServiceSettings: model.ServiceSettings{SiteURL: &siteURL}})
+
+	api.On("GetPost", "error_post").Return(nil, &model.AppError{Id: "1"})
+	api.On("GetPost", "post_not_found").Return(nil, (*model.AppError)(nil))
+
+	api.On("GetPost", "0").Return(&model.Post{UserId: "0"}, (*model.AppError)(nil))
+	api.On("GetUser", "0").Return(nil, &model.AppError{Id: "1"})
+
+	api.On("GetPost", "1").Return(&model.Post{UserId: "1"}, (*model.AppError)(nil))
+	api.On("GetUser", "1").Return(&model.User{Username: "username"}, (*model.AppError)(nil))
+
+	api.On("CreatePost", mock.AnythingOfType("*model.Post")).Return(nil, (*model.AppError)(nil))
+
+	p := Plugin{}
+	p.SetAPI(api)
+
+	p.userStore = getMockUserStoreKV()
+	p.currentInstanceStore = mockCurrentInstanceStore{&p}
+
+	tests := map[string]struct {
```

Comment:
I want to propose introducing the following struct that can be initialized for each subtest instead of creating the anonymous structs.
```go
	type requestStruct struct {
		PostId      string `json:"post_id"`
		CurrentTeam string `json:"current_team"`
		IssueKey    string `json:"issueKey"`
	}
```

Reply:
Done!